makefile.gen

```diff
@@ -81,11 +81,12 @@ release: $(OUT)/rom.bin $(OUT)/symbol.txt
 #release: $(info $$var is [${SRC_C}])
 
 debug: FLAGS= $(DEFAULT_FLAGS) -O1 -DDEBUG=1
-debug: CFLAGS= $(FLAGS) -ggdb
+debug: CFLAGS= $(FLAGS) -ggdb -g
 debug: AFLAGS= $(FLAGS)
 debug: LIBMD= $(LIB)/libmd_debug.a
 debug: $(OUT)/rom.bin $(OUT)/rom.out $(OUT)/symbol.txt
-.PHONY: debug
+debug: injectSymbolsInROM
+.PHONY: debug injectSymbolsInROM

```

Comment:
Not sure why you need to use
`.PHONY: debug injectSymbolsInRom`
instead of 
`.PHONY: debug`
as debug rules already requires it ?

Reply:
`.PHONY` in Makefiles has a special meaning as it defines a *[phony target](https://www.gnu.org/software/make/manual/make.html#Phony-Targets)*. It's a way to tell `make` "look, it's not a file name", otherwise you can create files named `debug`, `injectSymbolsInRom` etc and `make` won't make them if they exist and newer than their dependencies.

`debug: injectSymbolsInRom` won't mark a phony target just like `debug: rom.bin` doesn't, because we still can't tell it apart from a file with the same name.